src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,550 @@
+            return err;
+        }
+
+        return result;
+    }
+};
+
+# Worksheet Client Object.
+# + worksheetClient - HTTP client endpoint for the worksheet
+# + properties - worksheet specific properties
+# + microsoftGraphConfig - Configurations for accessing spreadsheet API
+public type Worksheet client object {
+    http:Client worksheetClient;
+    WorksheetProperties properties = {"path":"", "workbookName":"", "sheetId":"",
+                                                "worksheetName":"", "position":0};
+    MicrosoftGraphConfiguration microsoftGraphConfig;
+
+    public function __init(MicrosoftGraphConfiguration msGraphConfig, string path, string workbookName, string sheetId,
+                            string worksheetName, int position) {
+        self.microsoftGraphConfig = msGraphConfig;
+        self.properties = {"path":path, "workbookName":workbookName, "sheetId":sheetId,
+                            "worksheetName":worksheetName, "position":position};
+        oauth2:OutboundOAuth2Provider oauth2Provider3 = new ({
+            accessToken: msGraphConfig.msInitialAccessToken,
+            refreshConfig: {
+                clientId: msGraphConfig.msClientID,
+                clientSecret: msGraphConfig.msClientSecret,
+                refreshToken: msGraphConfig.msRefreshToken,
+                refreshUrl: msGraphConfig.msRefreshURL,
+                clientConfig: {
+                    secureSocket: {
+                        trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                        }
+                    }
+                }
+            }
+        });
+
+        http:BearerAuthHandler oauth2Handler3 = new (oauth2Provider3);
+
+        self.worksheetClient = new (msGraphConfig.baseUrl, {
+            auth: {
+                authHandler: oauth2Handler3
+            },
+            secureSocket: {
+                trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                }
+            }
+        });
+    }
+
+    # Get the properties of the Worksheet.
+    # + return - Properties of the Worksheet
+    public function getProperties() returns WorksheetProperties {
+        return self.properties;
+    }
+
+    # Create a new Table on this Worksheet.
+    # + tableName - name of the table to be created
+    # + address - The location where the table should be created
+    # + return - A Table client object on success, else returns an error
+    public remote function createTable(string tableName, string address) returns @tainted (Table|error) {
+        http:Request request = new;
+        json payload = {"name": tableName, "address": address, "hasHeaders": false};
+        request.setJsonPayload(payload);
```

Comment:
We can remove some of these variable declarations and pass the value in-line.

```suggestion
        request.setJsonPayload({"name": tableName, "address": address, "hasHeaders": false});
```

Reply:
Ok, Done